### PR TITLE
[JBPM-10051] Task operations fail intermittently when using LDAPUserGroupCallback

### PR DESCRIPTION
**Pull request by gmunozfe**
**JIRA**: [JBPM-10051](https://issues.redhat.com/browse/JBPM-10051)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by afalhambra** [comment]
Jenkins retest this

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/LDAPUserInfoImpl.java [hunk 1]
```diff
@@ -178,7 +178,7 @@ public String getEntityForEmail(String email) {
         String context = getConfigProperty(USER_CTX);
         String filter = getConfigProperty(EMAIL_FILTER);
         String attributeId = getConfigProperty(USER_ATTR_ID, DEFAULT_USER_ATTR_ID);
-        LdapSearcher searcher = ldapSearcher.search(context, filter, email);
+        final LdapResults searcher = ldapSearcher.search(context, filter, email);
```
Comment (afalhambra) [inline jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/LDAPUserInfoImpl.java:181]: sorry, I can't see why we need to declare it as `final`?

**Review by afalhambra — APPROVED** [review]
awesome job, well done @gmunozfe

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/LDAPUserInfoImpl.java (hunk 1 above)
Comment (elguardian) [reply jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/LDAPUserInfoImpl.java:181]: it does not any value a makes complex debugging.

**Review by elguardian — CHANGES_REQUESTED** [review]

File: jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/LDAPUserInfoImpl.java (hunk 1 above)
Comment (gmunozfe) [reply jbpm-human-task/jbpm-human-task-core/src/main/java/org/jbpm/services/task/identity/LDAPUserInfoImpl.java]: ok, removed final

**Comment by gmunozfe** [comment]
jenkins retest this

**Pull request merged**